### PR TITLE
Watch Face Updates

### PR DESCRIPTION
**Pull request by jomartigcal**
Using new GDG logo for the GDG watch face
Adding configurations for changing background, time display (12 hr/24 hr/hidden), and date/time color
Refactoring the setting/getting of watch face configurations

### CODE REVIEW

File: wear/src/main/java/org/gdg/frisbee/android/ColorConfigActivity.java [hunk 1]
```diff
@@ -61,16 +61,9 @@ public void onCentralPositionChanged(int i) {
     private void displayColorSelections() {
         List<String> colorList = new ArrayList<>();
 
-        if (mHeader.contains("Background")) {
-            String[] backgrounds = getResources().getStringArray(R.array.background_selection);
```
Comment (tasomaniac) [inline wear/src/main/java/org/gdg/frisbee/android/ColorConfigActivity.java]: Can you explain here? :)

File: wear/src/main/java/org/gdg/frisbee/android/GdgWatchFace.java [hunk 2]
```diff
@@ -99,14 +98,15 @@ public void handleMessage(Message message) {
...
         Bitmap mBackgroundBitmap;
-        Bitmap mDarkBackgroundBitmap;
-        Bitmap mLightBackgroundBitmap;
         Bitmap mGrayBackgroundBitmap;
+        int backgroundColor;
+        int timeSetting;
         boolean mAmbient;
         boolean mLightMode = false;
         boolean mDisplayDate = true;
+        boolean displayTime = true;
```
Comment (tasomaniac) [inline wear/src/main/java/org/gdg/frisbee/android/GdgWatchFace.java]: `displayTime` and `mDisplayDate` together does not look clean. 
Can you make them consistent?

File: wear/src/main/java/org/gdg/frisbee/android/ColorConfigActivity.java (hunk 1 above)
Comment (tasomaniac) [inline wear/src/main/java/org/gdg/frisbee/android/ColorConfigActivity.java]: I see, we no longer have background selection I guess.

File: wear/src/main/java/org/gdg/frisbee/android/GdgWatchFace.java [hunk 3]
```diff
@@ -151,9 +151,9 @@ public void onCreate(SurfaceHolder holder) {
             mBackgroundPaint = new Paint();
             mBackgroundPaint.setColor(ContextCompat.getColor(GdgWatchFace.this, R.color.gdg_black));
 
-            mBackgroundBitmap = BitmapFactory.decodeResource(resources, R.drawable.gdg_black);
```
Comment (tasomaniac) [inline wear/src/main/java/org/gdg/frisbee/android/GdgWatchFace.java]: Are we still using this drawable? We may remove it if we don't.

File: wear/src/main/java/org/gdg/frisbee/android/GdgWatchFace.java [hunk 4]
```diff
@@ -151,9 +151,9 @@ public void onCreate(SurfaceHolder holder) {
...
-            mDarkBackgroundBitmap = BitmapFactory.decodeResource(resources, R.drawable.gdg_black);
-            mLightBackgroundBitmap = BitmapFactory.decodeResource(resources, R.drawable.gdg_background);
+            mBackgroundBitmap = BitmapFactory.decodeResource(resources, R.drawable.gdg_new);
```
Comment (tasomaniac) [inline wear/src/main/java/org/gdg/frisbee/android/GdgWatchFace.java]: Can you remove `new` and just use something like `gdg_logo`?

File: wear/src/main/java/org/gdg/frisbee/android/GdgWatchFace.java [hunk 5]
```diff
@@ -209,37 +210,13 @@ public void onConnectionFailed(ConnectionResult connectionResult) {
...
-                            dataItems.release();
-                        }
-                    });
-        }
 
         @Override
         public void onDestroy() {
             mUpdateTimeHandler.removeMessages(R.id.message_update_time);
-
-            disconnectGoogleApiClient();
```
Comment (tasomaniac) [inline wear/src/main/java/org/gdg/frisbee/android/GdgWatchFace.java]: You are disconnecting in onVisibiltyChanged. I guess that is enough, right?

File: wear/src/main/java/org/gdg/frisbee/android/GdgWatchFace.java [hunk 6]
```diff
@@ -462,6 +438,16 @@ private Paint getAdjustedPaintColor(Paint paint) {
             return paint;
         }
 
+        private String formatHour(int hour) {
+            int hourToDisplay = hour;
+            if(timeSetting == WearableConfigurationUtil.TIME_12_HOUR) {
+                hourToDisplay = (hour % WearableConfigurationUtil.TIME_12_HOUR) == 0 ?
+                    WearableConfigurationUtil.TIME_12_HOUR : hour%WearableConfigurationUtil.TIME_12_HOUR;
```
Comment (tasomaniac) [inline wear/src/main/java/org/gdg/frisbee/android/GdgWatchFace.java]: There should be spaces around the 2nd `%`. I think we should enable checkstyle for this module too.

File: wear/src/main/java/org/gdg/frisbee/android/GdgWatchFaceConfigListenerService.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    public void onMessageReceived(MessageEvent messageEvent) {
+        super.onMessageReceived(messageEvent);
+
+        if (!messageEvent.getPath().equals(WearableConfigurationUtil.PATH_ANALOG)) {
+            return;
+        }
+
+        byte[] rawData = messageEvent.getData();
+        DataMap configKeys = DataMap.fromByteArray(rawData);
+        Log.d(TAG, "configKeys = [" + configKeys + "]");
```
Comment (tasomaniac) [inline wear/src/main/java/org/gdg/frisbee/android/GdgWatchFaceConfigListenerService.java]: We should log only on debug builds. Using Timber would be nice.

File: wear/src/main/java/org/gdg/frisbee/android/GdgWatchFaceConfigListenerService.java [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+        if (mGoogleApiClient == null) {
+            mGoogleApiClient = new GoogleApiClient.Builder(this).addConnectionCallbacks(this)
```
Comment (tasomaniac) [inline wear/src/main/java/org/gdg/frisbee/android/GdgWatchFaceConfigListenerService.java]: I guess there is no need to use `addConnectionCallbacks` here.

File: wear/src/main/java/org/gdg/frisbee/android/WearableConfigurationActivity.java [hunk 9]
```diff
@@ -86,31 +71,31 @@ public void onClick(WearableListView.ViewHolder viewHolder) {
 
                 TextView nameTextView = (TextView) layout.findViewById(R.id.setting_text_view);
                 String action = nameTextView.getText().toString();
-                if (action.contains("Background") || action.contains("Hand") || action.contains("Marker")) {
```
Comment (tasomaniac) [inline wear/src/main/java/org/gdg/frisbee/android/WearableConfigurationActivity.java]: We should avoid using String literals

**Comment by jomartigcal** [comment]
Hello! Will work on the fixes later.

File: wear/src/main/java/org/gdg/frisbee/android/ColorConfigActivity.java (hunk 1 above)
Comment (jomartigcal) [inline wear/src/main/java/org/gdg/frisbee/android/ColorConfigActivity.java]: I re-enabled background configuration. Changed background selection from dark/white to the same color selection for the others.

File: wear/src/main/java/org/gdg/frisbee/android/GdgWatchFaceConfigListenerService.java (hunk 8 above)
Comment (jomartigcal) [inline wear/src/main/java/org/gdg/frisbee/android/GdgWatchFaceConfigListenerService.java]: This is only used to log onConnected, onConnectionSuspended, and onConnectionFailed now. Should I remove it?

**Comment by friedger** [comment]
Could you fix the merge conflict?

**Comment by friedger** [comment]
:+1: LGTM

**Pull request merged**